### ISSUE TITLE
Implement replication_shutdown() to avoid relay crash at shutdown

### ISSUE THREAD

**Issue by kostja** [tarantool/tarantool#3485]
Relay threads keep using tx during shutdown, which leads to segmentation faults and aborts during shutdown.

How to reproduce (put this to reproduce file):
```
- [replication/autobootstrap.test.lua, memtx]                                   
- [replication/autobootstrap.test.lua, vinyl]                                   
- [replication/autobootstrap_guest.test.lua, memtx]                             
- [replication/autobootstrap_guest.test.lua, vinyl]                             
- [replication/before_replace.test.lua, memtx]                                  
- [replication/before_replace.test.lua, vinyl]                                  
- [replication/catch.test.lua, memtx]                                           
- [replication/catch.test.lua, vinyl]                                           
- [replication/consistent.test.lua, memtx]                                      
- [replication/consistent.test.lua, vinyl]                                      
- [replication/ddl.test.lua, memtx]                                             
- [replication/ddl.test.lua, vinyl]   
```

**Comment by locker** [comment]
For the record, another failure of the kind:
```
/src/cbus.c:210: cbus_endpoint_create: Assertion `e__ == 0' failed.
```

**Comment by locker** [comment]
Fixed by 4f8c988c717d8bf84c5be93cb6fc62c68fcb0b05
